src/Features/Core/Portable/ExtractMethod/MethodExtractor.cs

```diff
@@ -60,43 +64,60 @@ public async Task<ExtractMethodResult> ExtractMethodAsync(CancellationToken canc
 
             operationStatus = CheckVariableTypes(analyzeResult.Status.With(operationStatus), analyzeResult, cancellationToken);
             if (operationStatus.Failed())
-                return new FailedExtractMethodResult(operationStatus);
+                return ExtractMethodResult.Fail(operationStatus);
 
             var insertionPointNode = GetInsertionPointNode(analyzeResult, cancellationToken);
 
             if (!CanAddTo(originalSemanticDocument.Document, insertionPointNode, out var canAddStatus))
-                return new FailedExtractMethodResult(canAddStatus);
+                return ExtractMethodResult.Fail(canAddStatus);
 
             cancellationToken.ThrowIfCancellationRequested();
+            var codeGenerator = this.CreateCodeGenerator(analyzeResult);
 
-            var (analyzedDocument, insertionPoint) = await GetAnnotatedDocumentAndInsertionPointAsync(
-                originalSemanticDocument, analyzeResult, insertionPointNode, cancellationToken).ConfigureAwait(false);
+            var statements = codeGenerator.GetNewMethodStatements(insertionPointNode, cancellationToken);
+            if (statements.Status.Failed())
```

Comment:
>if (statements.Status.Failed()) [](http://example.com/codeflow?start=12&length=31)

General question: Does the HasRefactoringsAsync call into this still need to call in for both local/method or are they going to give the same result?

Reply:
they're different.  you can be offered one, or the other, or both.

HasRefactoring would still be good though as we could stop once we know about one.